src/mutation.h

```diff
@@ -3,20 +3,148 @@
 
 #include "pldata.h"
 #include "json.h"
-#include "enums.h" // tripoint
+#include "enums.h"
 #include "ui.h"
 #include <vector>
 #include <map>
 #include <unordered_map>
 
-struct dream;
-struct mutation_branch;
-
+extern std::map<std::string, mutation_type> mut_types;
 extern std::vector<dream> dreams;
-extern std::map<std::string, std::vector<std::string> > mutations_category;
-extern std::map<std::string, mutation_branch> mutation_data;
 typedef std::pair<body_part, tripoint> mutation_wet;
 
+/** Handles the "replaces" field for mutation_type. */
+void finalize_mut_types();
+
+enum mut_rating {
+    mut_good,     // A good mutation.
+    mut_bad,      // A bad mutation.
+    mut_mixed     // A mutation with both good and bad effects.
+    mut_neutral,  // A neutral or little effect mutation.
+}
+
+class mutation_type
+{
+        friend void load_mutation_type(JsonObject &jo);
+        friend void finalize_mut_types();
+        friend class mutation;
+    public:
+        // --------------- Constructors ---------------
+        mutation_type();
+        mutation_type(const mutation_type &rhs);
```

Comment:
Please, remove those copy constructors and let the compiler create them. You don't do anything special in it, so why implement it yourself. The compiler will create a version that is completely equivalent and it is much safer (the compiler won't forget to add code for a new member that is added later). And it's probably more efficient (at least not less efficient).

Similar, you might want to use in-class member initialization. It is less redundant (you don't need to do it for each constructor) and safer (change the initial value once and it is used in all constructors):
```C++
struct mutation {
    int visibility = 0;
};
```


Consider making `load_mutation_type` and the finalizer function public *static* members of `mutation_type`, they can be used the same as global functions, but they automatically have access to the class private members (no friend declaration required). See for example `material_type`. This allows to name the function consistently `mutation_type::load(JsonObject jo)`.